src/assembly/manifest.ts

```diff
@@ -0,0 +1,166 @@
+import * as path from 'path';
+import {
+    AssemblyManifest,
+    Manifest,
+    ArtifactType,
+    AwsCloudFormationStackProperties,
+    ArtifactMetadataEntryType,
+} from '@aws-cdk/cloud-assembly-schema';
+import * as fs from 'fs-extra';
+import { CloudFormationTemplate } from '../cfn';
+import { ArtifactManifest, AssetManifestProperties, LogicalIdMetadataEntry } from 'aws-cdk-lib/cloud-assembly-schema';
+import { CloudAssembly, CloudFormationStackArtifact } from 'aws-cdk-lib/cx-api';
+import { AssetManifest, DockerImageManifestEntry, FileManifestEntry } from 'cdk-assets';
+import { StackManifest } from './stack';
+import { ConstructTree, StackAsset, StackMetadata } from './types';
+
+/**
+ * Reads a Cloud Assembly manifest
+ */
+export class AssemblyManifestReader {
+    public static readonly DEFAULT_FILENAME = 'manifest.json';
+
+    /**
+     * Reads a Cloud Assembly manifest from a file
+     */
+    public static fromFile(fileName: string): AssemblyManifestReader {
+        try {
+            const obj = Manifest.loadAssemblyManifest(fileName);
+            return new AssemblyManifestReader(path.dirname(fileName), obj);
+        } catch (e: any) {
+            throw new Error(`Cannot read manifest '${fileName}': ${e.message}`);
+        }
+    }
+
+    /**
+     * Reads a Cloud Assembly manifest from a file or a directory
+     * If the given filePath is a directory then it will look for
+     * a file within the directory with the DEFAULT_FILENAME
+     */
+    public static fromPath(filePath: string): AssemblyManifestReader {
+        let st;
+        try {
+            st = fs.statSync(filePath);
+        } catch (e: any) {
+            throw new Error(`Cannot read manifest at '${filePath}': ${e.message}`);
+        }
+        if (st.isDirectory()) {
+            return AssemblyManifestReader.fromFile(path.join(filePath, AssemblyManifestReader.DEFAULT_FILENAME));
+        }
+        return AssemblyManifestReader.fromFile(filePath);
+    }
+
+    /**
+     * The directory where the manifest was found
+     */
+    public readonly directory: string;
+
+    private readonly assembly: CloudAssembly;
+    private readonly stacks = new Map<string, CloudFormationStackArtifact>();
```

Comment:
What is the string here? What's the map indexed on?

Compare for example this more spelled out form:

```
    /**
     * Map of resource logicalId to CloudFormation template resource fragment
     */
    private readonly resources: { [logicalId: string]: CloudFormationResource };
```

Might help us out later on.

Reply:
Realized that this actually isn't even needed so I removed it.